src/Microsoft.Crank.Controller/Ignore/IgnoreFile.cs

```diff
@@ -67,6 +58,28 @@ public static IgnoreFile Parse(string path, bool includeParentDirectories = fals
                                 continue;
                             }
 
+                            // Put a backslash ("\") in front of the first hash for patterns that begin with a hash.
```

Comment:
Nit: this comment looks like it was copied from the spec, our code removes the `\` which seems odd without context.
TLDR; add/update comment
Same in IgnoreRule.cs